### PR TITLE
ARROW-8458: [C++] Prefer the original mirrors for the bundled thirdparty dependencies

### PR DESCRIPTION
**Pull request by kszucs**
Also need to run the crossbow packaging tasks.

The recommended apache endpoint is documented [here](https://infra.apache.org/release-download-pages.html#closer).

### CODE REVIEW

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 1]
```diff
@@ -264,16 +264,15 @@ else()
                  ${ARROW_BOOST_BUILD_VERSION})
   set_urls(
     BOOST_SOURCE_URL
+    "https://dl.bintray.com/boostorg/release/${ARROW_BOOST_BUILD_VERSION}/source/boost_${ARROW_BOOST_BUILD_VERSION_UNDERSCORES}.tar.gz"
```
Comment (bkietz) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake:267]: Please prefer the trimmed boost bundles over full boost bundles

**Review by bkietz — CHANGES_REQUESTED** [review]

File: cpp/cmake_modules/ThirdpartyToolchain.cmake [hunk 2]
```diff
@@ -264,16 +264,15 @@ else()
...
+    "https://github.com/boostorg/boost/archive/boost-${ARROW_BOOST_BUILD_VERSION}.tar.gz"
```
Comment (nealrichardson) [inline cpp/cmake_modules/ThirdpartyToolchain.cmake:268]:
```suggestion
```

And if you want to bump the others up, put them after the ursalabs/arrow-boost one.

File: cpp/cmake_modules/ThirdpartyToolchain.cmake (hunk 1 above)
Comment (kszucs) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake:267]: You mean "https://dl.bintray.com/ursalabs/arrow-boost/boost_${ARROW_BOOST_BUILD_VERSION_UNDERSCORES}.tar.gz" ?

Comment (nealrichardson) [reply cpp/cmake_modules/ThirdpartyToolchain.cmake:267]: Yes, please see the code comment that explains what it is. See also ARROW-8222.

**Comment by kszucs** [comment]
@github-actions crossbow submit -g nightly

**Review by kszucs — COMMENTED** [review]
+1, the relevant builds are passing

**Pull request closed without merging**